emulator/src/core/emulator/instructions/move.rs

```diff
@@ -0,0 +1,124 @@
+use super::super::{Emulator, EResult};
+
+const REGISTERS: [char; 8] = ['b', 'c', 'd', 'e', 'h', 'l', 'm', 'a'];
+
+impl Emulator {
+    pub fn mvi(&mut self, r: char) -> EResult<()> {
+        self.reg[r] = self.read_byte()?;
+        Ok(())
+    }
+
+    pub fn mvi_adr(&mut self) -> EResult<()> {
+        // Move byte 2 to address in HL
+        let byte = self.read_byte()?;
+        let adr = self.reg["hl"];
+        self.ram[adr] = byte;
+        Ok(())
+    }
+
+    pub fn resolve_mov(&mut self, opcode: u8) -> EResult<()> {
+        let opcode_rel = opcode - 0x40;
+        let dst_idx = opcode_rel >> 3;
+        let src_idx = opcode_rel - (dst_idx << 3);
+        if dst_idx == 6 {
+            self.ram[self.reg["hl"]] = self.reg[REGISTERS[src_idx as usize]];
+        } else {
+            if src_idx == 6 {
+                self.reg[REGISTERS[dst_idx as usize]] = self.ram[self.reg["hl"]];
+            } else {
+                self.mov(REGISTERS[dst_idx as usize], REGISTERS[src_idx as usize])?;
+            }
+        }
+        Ok(())
+    }
+
+    pub fn mov(&mut self, dst: char, src: char) -> EResult<()> {
+        self.reg[dst] = self.reg[src];
+        Ok(())
+    }
+
+    pub fn lxi(&mut self, dst: &str) -> EResult<()> {
+        self.reg[dst] = self.read_addr()?;
+        Ok(())
+    }
+}
+
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    
+    
+    use crate::core::emulator::Emulator;
+    use crate::kreator::assembler::Assembler;
+    use std::{
+        fs::*,
+        io::{self, Read},
+    };
+
+    fn load_asm_file(emulator: &mut Emulator, path: &str) -> io::Result<()> {
```

Comment:
We should probably move this function to some kind of utils file, I guess... not sure where exactly though.

Reply:
Maybe `src/core/emulator/utils.rs` or something like that

Reply:
Can't access "ram" there (private member)